src/main/java/org/broadinstitute/hellbender/utils/read/markduplicates/ReadsKey.java

```diff
@@ -101,48 +76,51 @@ public String toString() {
     }
 
     /**
+     * Key class for representing relevant duplicate marking identifiers into a two long key values for pair data data.
      *
+     * Note: This class is intended for internal MarkDuplicatesSpark key purposes, it is only exposed so it can
+     *       be accessed by {@link org.broadinstitute.hellbender.engine.spark.GATKRegistrator} for kryo serialization
      */
-    public static class keyForPair extends ReadsKey {
-        final long fragmentValue;
-        final long keyValue;
+    public static class KeyForPair extends ReadsKey {
+        final long firstReadKeyValue;
+        final long secondReadKeyValue;
 
-        keyForPair(final long fragmentValue, final long pairValue) {
-            this.fragmentValue = fragmentValue;
-            this.keyValue = pairValue;
+        KeyForPair(final long firstReadKeyValue, final long secondReadKeyValue) {
+            this.firstReadKeyValue = firstReadKeyValue;
+            this.secondReadKeyValue = secondReadKeyValue;
         }
 
         @Override
         public boolean equals(Object o) {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
-            keyForPair that = (keyForPair) o;
-            return fragmentValue == that.fragmentValue &&
-                    keyValue == that.keyValue;
+            KeyForPair that = (KeyForPair) o;
+            return firstReadKeyValue == that.firstReadKeyValue &&
+                    secondReadKeyValue == that.secondReadKeyValue;
         }
 
         @Override
         public int hashCode() {
-
-            return Objects.hash(fragmentValue, keyValue);
+            return Objects.hash(firstReadKeyValue, secondReadKeyValue);
         }
 
         @Override
         public String toString() {
-            return fragmentValue + " " + keyValue;
+            return firstReadKeyValue + " " + secondReadKeyValue;
         }
     }
 
-
+    // Helper methods for generating summary longs
     private static long longKeyForFragment(int strandedUnclippedStart, boolean reverseStrand, int referenceIndex, byte library) {
         return (((long)strandedUnclippedStart) << 32) |
```

Comment:
could you add a comment explaining the reason you need to mask these, it wasn't obvious to us at first, so it probably won't be when we've come back to look at it later.